File: desktop/windows/VoiceBridgeAI/VoiceBridgeAI/App/BundleVariant.cs
namespace VoiceBridgeAI;

/// <summary>Build flavor: cloud | local | standard (mirrors macOS BundleVariant).</summary>
public static class BundleVariant
{
    public enum Kind
    {
        Cloud,
        Local,
        Standard,
    }

    public static Kind Current { get; } = Resolve();

    public static bool IncludesLocalModels => Current != Kind.Cloud;

    public static string AppSupportFolderName => Current switch
    {
        Kind.Cloud => "VoiceBridgeAI-Cloud",
        Kind.Local => "VoiceBridgeAI-Local",
        _ => "VoiceBridgeAI",
    };

    public static string DisplaySuffix => Current switch
    {
        Kind.Cloud => "（云端）",
        Kind.Local => "（本地）",
        _ => "",
    };

    private static Kind Resolve()
    {
        var fromEnv = Environment.GetEnvironmentVariable("VOICEBRIDGE_BUNDLE_VARIANT");
        if (TryParse(fromEnv, out var envKind))
        {
            return envKind;
        }

        var marker = Path.Combine(AppContext.BaseDirectory, "bundle-variant.txt");
        if (File.Exists(marker))
        {
            var text = File.ReadAllText(marker).Trim();
            if (TryParse(text, out var fileKind))
            {
                return fileKind;
            }
        }

        return Kind.Standard;
    }

    private static bool TryParse(string? raw, out Kind kind)
    {
        kind = Kind.Standard;
        if (string.IsNullOrWhiteSpace(raw))
        {
            return false;
        }

        return raw.Trim().ToLowerInvariant() switch
        {
            "cloud" => Assign(Kind.Cloud, out kind),
            "local" => Assign(Kind.Local, out kind),
            _ => false,
        };
    }

    private static bool Assign(Kind value, out Kind kind)
    {
        kind = value;
        return true;
    }
}

File: desktop/macos/Sources/VoiceBridgeAI/App/BundleVariant.swift
import Foundation

/// Build flavor written into Info.plist by `build-app.sh` (`cloud` | `local`).
enum BundleVariant {
    enum Kind: String {
        case cloud
        case local
        case standard
    }

    static var current: Kind {
        guard let raw = Bundle.main.infoDictionary?["VoiceBridgeBundleVariant"] as? String else {
            return .standard
        }
        return Kind(rawValue: raw.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()) ?? .standard
    }

    static var includesLocalModels: Bool {
        current != .cloud
    }

    static var appSupportFolderName: String {
        switch current {
        case .cloud: return "VoiceBridgeAI-Cloud"
        case .local: return "VoiceBridgeAI-Local"
        case .standard: return "VoiceBridgeAI"
        }
    }

    static var displaySuffix: String {
        switch current {
        case .cloud: return "（云端）"
        case .local: return "（本地）"
        case .standard: return ""
        }
    }
}
